hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/RegionServerRpcQuotaManager.java

```diff
@@ -157,39 +157,48 @@ public OperationQuota getQuota(final UserGroupInformation ugi, final TableName t
   /**
    * Check the quota for the current (rpc-context) user. Returns the OperationQuota used to get the
    * available quota and to report the data/usage of the operation.
-   * @param region the region where the operation will be performed
-   * @param type   the operation type
+   * @param region   the region where the operation will be performed
+   * @param type     the operation type
+   * @param isAtomic whether the given operation is atomic
    * @return the OperationQuota
    * @throws RpcThrottlingException if the operation cannot be executed due to quota exceeded.
    */
-  public OperationQuota checkQuota(final Region region, final OperationQuota.OperationType type)
-    throws IOException, RpcThrottlingException {
+  public OperationQuota checkQuota(final Region region, final OperationQuota.OperationType type,
```

Comment:
I'm thinking we should add another OperationType CHECK_AND_MUTATE. We can pass that in here instead of adding an isAtomic boolean.  We can also add a util in QuotaUtil to return an OperationType for an Action + hasCondition boolean. 

We can then use that same api in the other checkQuota below. Something like:
```java
for (final ClientProtos.Action action : actions) {
  OperationType type = QuotaUtil.getOperationType(action, hasCondition);
  if (type.isRead()) {
    numReads++;
  }
  if (type.isWrite()) {
    numWrites++;
  }
}
```

The goal with this change is to DRY up the mutation type checking, and also to clarify a bit. Currently we are using the `isAtomic` boolean name which I'm realizing is not accurate. `RowMutations` is an example of an atomic operation which is not conditional (thus has no reads).